lib/rails/observers/active_resource/observing.rb

```diff
@@ -26,6 +27,7 @@ def update(*)
       notify_observers(:before_update)
       if result = super
         notify_observers(:after_update)
+        notify_observers(:after_update_commit)
```

Comment:
This makes me wonder if we should even support `after_*_commit`. They aren't really any different from `after_update` and friends, so why try to make the distinction that we support them?

@rafaelfranca thoughts?

Reply:
@kaspth Support for `after_create_commit` makes a big difference in our application, hence the fork. I maintain a strict separation of concerns in code:

* Model callbacks affect internal state and child records _only_.
* Observer callbacks enqueue ActiveJob tasks which perform complex//third-party connections.

I pass entire objects (see [GlobalID](http://edgeguides.rubyonrails.org/active_job_basics.html#globalid)) to ActiveJob. GlobalID takes the class and ID, and fetches it from the database. If the object doesn't yet exist in the database (possible when `after_create` fires), ActiveJob will raise `ActiveJob::DeserializationError` and push the task back onto the queue for later execution. This leads to an unacceptable delay in message delivery and notification broadcast to our customers.

`after_create_commit` fires after the record has been written to the database, and for the purposes of ActiveJob it's the safest callback to target on record creation.